# Imports

In [1]:
import pandas as pd
import os

# Reading Data

In [2]:
df = pd.read_csv('sales.csv')
cols = df.columns
cols

Index(['invoice_and_item_number', 'date', 'store_number', 'store_name',
       'address', 'city', 'zip_code', 'store_location', 'county_number',
       'county', 'category', 'category_name', 'vendor_number', 'vendor_name',
       'item_number', 'item_description', 'pack', 'bottle_volume_ml',
       'state_bottle_cost', 'state_bottle_retail', 'bottles_sold',
       'sale_dollars', 'volume_sold_liters', 'volume_sold_gallons'],
      dtype='object')

# Exploratory Data Analysis

In [3]:
print(f'Total number of records: {len(df)}')

Total number of records: 2805307


In [4]:
df.isnull().sum()

invoice_and_item_number         0
date                            0
store_number                    0
store_name                      0
address                       142
city                          142
zip_code                      142
store_location             330335
county_number                 142
county                        142
category                        0
category_name                   0
vendor_number                   4
vendor_name                     4
item_number                     0
item_description                0
pack                            0
bottle_volume_ml                0
state_bottle_cost               0
state_bottle_retail             0
bottles_sold                    0
sale_dollars                    0
volume_sold_liters              0
volume_sold_gallons             0
dtype: int64

## Remove Store Location

There are a lot of entries missing for the store locations. We are not interested in the exact store location anyways. The county is enough!

In [5]:
df.drop('store_location', axis=1, inplace=True)
cols = df.columns
cols

Index(['invoice_and_item_number', 'date', 'store_number', 'store_name',
       'address', 'city', 'zip_code', 'county_number', 'county', 'category',
       'category_name', 'vendor_number', 'vendor_name', 'item_number',
       'item_description', 'pack', 'bottle_volume_ml', 'state_bottle_cost',
       'state_bottle_retail', 'bottles_sold', 'sale_dollars',
       'volume_sold_liters', 'volume_sold_gallons'],
      dtype='object')

In [6]:
df.isnull().sum()

invoice_and_item_number      0
date                         0
store_number                 0
store_name                   0
address                    142
city                       142
zip_code                   142
county_number              142
county                     142
category                     0
category_name                0
vendor_number                4
vendor_name                  4
item_number                  0
item_description             0
pack                         0
bottle_volume_ml             0
state_bottle_cost            0
state_bottle_retail          0
bottles_sold                 0
sale_dollars                 0
volume_sold_liters           0
volume_sold_gallons          0
dtype: int64

Still we have 142 missing values. Let's hope they 'overlap'!

In [7]:
df.dropna(inplace=True)

In [8]:
df.isnull().sum()

invoice_and_item_number    0
date                       0
store_number               0
store_name                 0
address                    0
city                       0
zip_code                   0
county_number              0
county                     0
category                   0
category_name              0
vendor_number              0
vendor_name                0
item_number                0
item_description           0
pack                       0
bottle_volume_ml           0
state_bottle_cost          0
state_bottle_retail        0
bottles_sold               0
sale_dollars               0
volume_sold_liters         0
volume_sold_gallons        0
dtype: int64

Hurray! We don't have any more missing values...

In [9]:
print(f'Total number of records: {len(df)}')

Total number of records: 2805161


# Splitting Files into Year and Month

In [10]:
df['Year'] = df['date'].apply(lambda x: x.split('-')[0])
df['Month'] = df['date'].apply(lambda x: x.split('-')[1])

In [11]:
set(df['Year']), set(df['Month'])

({'2021', '2022'},
 {'01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12'})

In [12]:
for i in set(df.Year): 
    for j in set(df.Month):
        filename = "liquor_"+i+"_"+j+".csv"
        df.loc[df.Year == i][df.Month == j].to_csv(filename,index=False,columns=cols)

/tmp/ipykernel_19854/2156608742.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  df.loc[df.Year == i][df.Month == j].to_csv(filename,index=False,columns=cols)


# Verifying that the total number of records matches that in the split files

In [13]:
total_df = 0
for i in set(df.Year): 
    for j in set(df.Month):
        filename = "liquor_"+i+"_"+j+".csv"
        temp_df = pd.read_csv(filename)
        len_df = len(temp_df)
        if len_df == 0:
            os.remove(filename)
        total_df += len_df
print(f'Total number of records in split files: {total_df}')

Total number of records in split files: 2805161
